node-drainer/pkg/queue/queue.go

```diff
@@ -56,18 +56,53 @@ func (m *eventQueueManager) EnqueueEventGeneric(ctx context.Context,
 	default:
 	}
 
+	// Extract eventID from the event map
+	eventID := extractEventID(event)
+
 	nodeEvent := NodeEvent{
 		NodeName: nodeName,
+		EventID:  eventID,
 		Event:    &event,
 		Database: database,
 	}
 
+	slog.Debug("Enqueueing event", "nodeName", nodeName, "eventID", eventID)
+
 	m.queue.Add(nodeEvent)
 	metrics.QueueDepth.Set(float64(m.queue.Len()))
 
 	return nil
 }
 
+// extractEventID extracts the document ID from an event map
+// Tries both MongoDB (_id) and PostgreSQL (id) formats
+func extractEventID(event datastore.Event) string {
```

Comment:
Should we move this into the datastore package so that client don't have to be aware of the datatsore specific logic?

Reply:
good call. switching to another method we have already in store-client.